src/_pytest/cacheprovider.py

```diff
@@ -276,7 +276,11 @@ def pytest_collection_modifyitems(self, session, config, items):
             items[:] = self._get_increasing_order(
                 new_items.values()
             ) + self._get_increasing_order(other_items.values())
-        self.cached_nodeids = [x.nodeid for x in items if isinstance(x, pytest.Item)]
+        else:
+            for item in items:
+                if item.nodeid not in self.cached_nodeids:
+                    new_items[item.nodeid] = item
+        self.cached_nodeids.extend(new_items)
```

Comment:
With regard to dropping `isinstance` here: see https://github.com/pytest-dev/pytest/pull/3218#discussion_r168623666
Should either be kept if we have a test for this (`testing/acceptance_test.py` worked for me), or have a test to cover it if necessary.

Reply:
Fair enough!